### DISCUSSION TITLE
Pan/Tilt speed and zoom level values in ONVIF

### DISCUSSION THREAD

**Discussion by xtrdustin** [AxisCommunications/discussions#80] [General] [answered]
There are some values in the ONVIF PTZ that are, from what I can tell, implementation defined. Can you tell me what these correspond to in the Axis Q6215-LE camera (or point me to documentation that would have the corresponding actual values for this camera)?

- One that we are interested in is the zoom level. In ONVIF, "Generic Zoom Position Space" goes from 0 to 1, and we are relating it to field of view in degrees, but we don't know the actual field of view, which would help us calculate how much to pan or tilt the camera, when we want to point the camera at a specific point on the screen.
- The other is the velocity component of the move requests. The X and Y components both go from -1 to 1 in "Generic Pan/Tilt Velocity Space".  For a continuous move command, we are converting from degrees per second for azimuth and elevation, but don't know how that corresponds to the -1 to 1 values (and what the min/max pan/tilt of the camera are).

It's possible I'm just missing something in the ONVIF spec, but any info on these would be appreciated.

Thanks!

**Comment by mattias-kindborg-at-work** [comment]
Hi @xtrdustin. Please look into using the relative pan/tilt translation space in FoV for the case where you want to point the camera at a specific point on the screen.

Unfortunately, we currently do not support continuous movements using the pan/tilt velocity space in degrees per second in Onvif. The product specific information on speed  values can be obtained from the camera using a query that provides a mapping table of generic speed in the range {1,100} to speed values in degrees per second, see `http://<ip>/axis-cgi/com/ptz.cgi?query=attributes`.

**Reply by xtrdustin** [reply]
Thanks for getting back to me.

We don't need to send degrees per second, we just need to know the meaning of the units to send the speed in so that we can convert to them from our software which uses degrees per second internally.  I'll check out the information at this URL when we have our camera back from the team that's currently borrowing it (hopefully in the next few days).

Regarding the FOV range, we found that in the [Technical Specs tab here](https://www.axis.com/products/axis-q6215-le#technical-specifications):

* Horizontal field of view: 58.6 - 2.2 °
* Vertical field of view: 34.1 - 1.2 °

**Reply by xtrdustin** [reply, marked as answer]
@mattias-kindborg-at-work We'd like some help understanding how this line translates to ONVIF:

`speeds=value=1|speed=0.05,value=4|speed=0.06,value=9|speed=0.16,value=11|speed=0.25,value=14|speed=2,value=19|speed=0.46,value=24|speed=2.1,value=31|speed=4.5,value=40|speed=9,value=50|speed=19,value=58|speed=28,value=64|speed=38,value=80|speed=74,value=86|speed=92,value=100|speed=150`

I interpret this as: 1 means 0.05 degrees per second and 100 means 150 deg/sec. 

However, ONVIF seems to take values from 0 to 1. Looking at the PTZ Service Specification, the section on Generic Speed Space says that it's normalized from 0=stopped to 1=full speed.  So I guess that would mean the ONVIF value 0.01 corresponds to 1 = .05 deg/sec and ONVIF value 1 corresponds to 100 = 150 deg/sec?

**Reply by mdaconta** [reply]
In looking at the speed table returned by Vapix, there appears to be an anomaly at value=14.
speeds=value=1|speed=0.05,value=4|speed=0.06,value=9|speed=0.16,value=11|speed=0.25,value=14|speed=2,value=19|speed=0.46,value=24|speed=2.1,value=31|speed=4.5,value=40|speed=9,value=50|speed=19,value=58|speed=28,value=64|speed=38,value=80|speed=74,value=86|speed=92,value=100|speed=150

Looking at the speed for the previous point is (11,0.25) and the following point is (19,0.45)... it seems like an error that value 14 would have a speed of 2.  Such a "spike" in the graph is suspicious.  Is that a bug in VAPIX?   How is this table generated?
